### PR TITLE
Add provider health status event

### PR DESCRIPTION
**Pull request by giusdp**
## Feature or Problem
From the discussions of the following issues, providers should emit the health events on the `wasmbus.evt` topic.
Please let me know if I misunderstood what was being asked in those issues or if there is something missing.

## Related Issues
- #517 
- #519 

## Release Information
Next

## Consumer Impact
N/A

## Testing

Built on platform(s)
- [x] x86_64-linux

Tested on platform(s)
- [x] x86_64-linux

### Unit Test(s)
I've added a test in test/host_core/provider_tests.exs to start a provider from file and wait on the health_check events. It first waits for the health_check_passed event when the provider is started and afterwards waits for the health_check_status event once, which is emitted every 30 seconds.

### Manual Verification

I've launched nats and subscribed to `wasmbus.evt.*` and to ">". 
I've used `make run` from the dashboard project and from the web page started a provider from file to checkout that the health_status event was being correctly sent.

### CODE REVIEW

File: host_core/lib/host_core/providers/provider_module.ex [hunk 1]
```diff
@@ -395,6 +395,7 @@ defmodule HostCore.Providers.ProviderModule do
           %State{state | healthy: true}
 
         {:ok, _body} ->
+          publish_health_status(state)
```
Comment (connorsmith256) [inline host_core/lib/host_core/providers/provider_module.ex:398]: If we're always going to publish status events, I think this block should be refactored. Something like:
```elixir
new_state = case res do
  {:ok, _body} when not state.healthy ->
    %State(state | healthy: true}
  {:error, _} when state.healthy ->
    %State(state | healthy: false}
  _ ->
    state
end
publish_health_status(new_state)
{:noreply, new_state}
```

(`publish_health_status` might need to change too, to infer what the name of the cloud event should be from whether the state is healthy or not)

This separates updating state, which we want to do when the health status changes, from publishing, which we want to do every time now

**Review by connorsmith256 — CHANGES_REQUESTED** [review]
Thank you for another contribution @giusdp !

I have one request to make the code more readable

File: host_core/lib/host_core/providers/provider_module.ex (hunk 1 above)
Comment (giusdp) [reply host_core/lib/host_core/providers/provider_module.ex]: Thanks for the suggestion! I refactored the code and changes `publish_health_status(new_state)` to handle the cases for the passed, failed health event and the normal status event.

**Review by connorsmith256 — APPROVED** [review]
LGTM!

**Comment by connorsmith256** [comment]
I'm going to hold off merging for now since we're hoping to cut a release for v0.61 this week, but thanks again!

**Comment by giusdp** [comment]
No problem!

**Pull request merged**